js/components/Tabs.js

```diff
@@ -0,0 +1,38 @@
+const React = require("react");
+const ReactDOM = require("react-dom");
+const { bindActionCreators } = require("redux");
+const { connect } = require("react-redux");
+const { getTabs } = require("../queries");
+const actions = require("../actions")
+
+require("./tabs.css");
+const dom = React.DOM;
+
+function Tabs({tabs, selectTab, loadSources}) {
+  const tabsArr = Object.keys(tabs).map(k => tabs[k]);
+
+  function onClickTab(e) {
+    selectTab({ tabActor: e.currentTarget.dataset.actorId })
+      .then(loadSources);
```

Comment:
Yeah, it's generally not good to tie down actions like that, so I think this is fine. I don't think it's really that dirty: it's not uncommon to want to hook up a workflow like this, and that's why any async actions that use the promise middleware return promises when they are dispatched. It allows you to do stuff like this.